core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java

```diff
@@ -130,7 +142,7 @@ public class DateTimeFormatters {
 
   public static final DateTimeFormatter DATE_TIME_FORMATTER_VARIABLE_NANOS_OPTIONAL =
       new DateTimeFormatterBuilder()
-          .appendPattern("[uuuu-MM-dd HH:mm:ss][uuuu-MM-dd HH:mm][HH:mm:ss][HH:mm][uuuu-MM-dd]")
+          .appendPattern("[" + String.join("][", DATE_TIME_FORMATTER_PATTERNS_NANOS_OPTIONAL) + "]")
```

Comment:
Not a blocker, but it is better doing like wrapping each item like `"[" + item + "]"` or `String.format("[%s]")` and then join without delimiter. It would make it more straight forward and clearer.

Reply:
This is not needed as I'm using OpenSearch Date formatters to parse the date string.